internal/operator-controller/applier/helm.go

```diff
@@ -164,9 +165,30 @@ func (h *Helm) buildHelmChart(bundleFS fs.FS, ext *ocv1.ClusterExtension) (*char
 	return h.BundleToHelmChartFn(bundleFS, ext.Spec.Namespace, watchNamespace)
 }
 
-func (h *Helm) getReleaseState(cl helmclient.ActionInterface, ext *ocv1.ClusterExtension, chrt *chart.Chart, values chartutil.Values, post postrender.PostRenderer) (*release.Release, *release.Release, string, error) {
+func (h *Helm) getReleaseState(ctx context.Context, cl helmclient.ActionInterface, ext *ocv1.ClusterExtension, chrt *chart.Chart, values chartutil.Values, post postrender.PostRenderer) (*release.Release, *release.Release, string, error) {
+	logger := log.FromContext(ctx)
 	currentRelease, err := cl.Get(ext.GetName())
+
+	// if a release is pending at this point, that means that a helm action
+	// (installation/upgrade) we were attempting was likely interrupted in-flight.
+	// Pending release would leave us in reconciliation error loop because helm
+	// wouldn't be able to progress automatically from it.
+	//
+	// one of the workarounds is to try and remove helm metadata relating to
+	// that pending release which should 'reset' its state communicated to helm
+	// and the next reconciliation should be able to successfully pick up from here
+	// for context see: https://github.com/helm/helm/issues/5595 and https://github.com/helm/helm/issues/7476
+	// and the discussion in https://github.com/operator-framework/operator-controller/pull/1776
+	if err == nil && currentRelease.Info.Status.IsPending() {
+		if _, err = cl.Config().Releases.Delete(currentRelease.Name, currentRelease.Version); err != nil {
+			return nil, nil, StateError, fmt.Errorf("failed removing interrupted release %q version %d metadata: %w", currentRelease.Name, currentRelease.Version, err)
+		}
+		// return error to try to detect proper state (installation/upgrade) at next reconciliation
+		return nil, nil, StateError, fmt.Errorf("removed interrupted release %q version %d metadata", currentRelease.Name, currentRelease.Version)
```

Comment:
That is the way that we all discussed in the team, right? 
So, we delete the latest release and try again 👍

Reply:
yes, we discussed this in the team among the option to rollback and manually change the release state.
The idea to choose to go with deleting latest release comes mostly from this thread: https://github.com/operator-framework/operator-controller/pull/1776#discussion_r1958491722 and the changes we did in `helm-operator-plugins` to be able to facilitate this solution (https://github.com/operator-framework/helm-operator-plugins/pull/424)

Reply:
IHMO that is a good / safe option 👍 
I am OK with